manifests/job.pp

```diff
@@ -34,11 +37,16 @@
   $jobname  = $title,
   $enabled  = 1,
   $ensure   = 'present',
+  $folder   = undef,
   $difftool = '/usr/bin/diff -b -q',
 ){
   include ::jenkins::cli
 
-  validate_string($difftool)
+  validate_string($difftool, $folder)
+
+  if $folder and (!defined(Jenkins::Plugin['cloudbees-folder'])) {
+    fail('cloudbees-folder plugin is required for the \'folder\' parameter.')
+  }
```

Comment:
@carroarmato0 I wasn't suggestion that the plugin be automatically installed -- I actually would prefer that not happen.

Note that `defined()` is parse order dependent AND it does not create an ordering relationship in the catalog.  If it's necessary to constrain in this way, it would be better to use the plugin fact but it would likely mean it would take multiple runs to to converge a catalog that both installs the plugin and a folder job.

What I was suggesting is that a dep between the sub-job and the parent folder be automatically declared instead of requiring the end-user to declare this relationship as you have done in your example manifest.